src/mscorlib/src/System/String.cs

```diff
@@ -411,7 +415,61 @@ private unsafe static bool EqualsHelper(String strA, String strB)
                 return (length <= 0);
             }
         }
-        
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        private unsafe static bool OrdinalCompareSubstring(String strA, String strB, int length)
+        {
+            Contract.Requires(strA != null);
+            Contract.Requires(strB != null);
+            Contract.Requires(strA.Length <= length);
+            Contract.Requires(strB.Length <= length);
+
+            fixed (char* ap = &strA.m_firstChar) fixed (char* bp = &strB.m_firstChar)
+            {
+                char* a = ap;
+                char* b = bp;
+
+#if AMD64
+#if ALIGN_ACCESS
+    #error Needs porting for ALIGN_ACCESS (https://github.com/dotnet/corert/issues/430) 
+#endif // ALIGN_ACCESS
```

Comment:
The fix for aligned access is trivial here - do one int comparison before starting the unrolled loop (this assumes that the caller has the fast path for single character), so I would just do it.

I am wondering whether it should be on all the time, without ifdef. The aligned access is general goodness (even if it does not show up in a particular micro-benchmark).